StreamVideoTests/CallKit/CallKitServiceTests.swift

```diff
@@ -0,0 +1,197 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import CallKit
+import Foundation
+@testable import StreamVideo
+import XCTest
+
+final class CallKitServiceTests: XCTestCase {
+
+    private lazy var subject: CallKitService! = .init()
+    private lazy var callController: MockCXCallController! = .init()
+    private lazy var callProvider: MockCXProvider! = .init()
+    private lazy var cid: String = "default:\(callId)"
+
+    private var callId: String = String(UUID().uuidString.replacingOccurrences(of: "-", with: "").prefix(10))
+    private var localizedCallerName: String = "Test Caller"
+    private var callerId: String = "test@example.com"
+
+    override func setUp() {
+        super.setUp()
+        subject.callController = callController
+        subject.callProvider = callProvider
+    }
+
+    override func tearDown() {
+        subject = nil
+        super.tearDown()
+    }
+
+    // MARK: - reportIncomingCall
+
+    func test_reportIncomingCall_callProviderWasCalledWithExpectedValues() {
+        // Given
+        let expectation = self.expectation(description: "Report Incoming Call")
+        var completionError: Error?
+
+        // When
+        subject.reportIncomingCall(
+            cid,
+            localizedCallerName: localizedCallerName,
+            callerId: callerId
+        ) { error in
+            completionError = error
+            expectation.fulfill()
+        }
+
+        // Then
+        waitForExpectations(timeout: defaultTimeout, handler: nil)
+        XCTAssertNil(completionError)
+        XCTAssertTrue(callProvider.reportNewIncomingCallCalled)
+        XCTAssertEqual(callProvider.reportNewIncomingCallUpdate?.localizedCallerName, localizedCallerName)
+        XCTAssertEqual(callProvider.reportNewIncomingCallUpdate?.remoteHandle?.value, callerId)
+    }
+
+    func test_reportIncomingCall_streamVideoIsNil_callWasEnded() async throws {
+
```

Comment:
Why do we have empty space before each of these tests? And why is not the linter complaining 😅